File: frontend/src/stores/project.ts
/**
 * Project Store - 项目（IP）状态管理
 * 
 * 使用 Pinia 管理多项目切换和当前激活项目
 * 支持本地持久化存储
 */

import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
import { get, post, put, del } from '@/utils/request'

// ============== 类型定义 ==============

// 人设配置类型
export interface PersonaSettings {
  tone: string
  catchphrase: string
  target_audience: string
  benchmark_accounts: string[]
  content_style: string
  taboos: string[]
  keywords: string[]
  introduction: string
}

// 项目类型
export interface Project {
  id: string
  user_id: string
  name: string
  industry: string
  avatar_letter: string
  avatar_color: string
  persona_settings: PersonaSettings
  created_at: string
  updated_at: string
  is_active: boolean
}

// 创建项目请求
export interface ProjectCreateRequest {
  name: string
  industry?: string
  persona_settings?: Partial<PersonaSettings>
}

// 更新项目请求
export interface ProjectUpdateRequest {
  name?: string
  industry?: string
  persona_settings?: Partial<PersonaSettings>
}

// 缓存 key
const ACTIVE_PROJECT_KEY = 'sfire_ai_active_project'
const PROJECT_LIST_KEY = 'sfire_ai_project_list'

// 默认人设配置
export const DEFAULT_PERSONA_SETTINGS: PersonaSettings = {
  tone: '专业亲和',
  catchphrase: '',
  target_audience: '',
  benchmark_accounts: [],
  content_style: '',
  taboos: [],
  keywords: [],
  introduction: ''
}

/**
 * Project Store
 */
export const useProjectStore = defineStore('project', () => {
  // ============== State ==============
  
  // 项目列表
  const projectList = ref<Project[]>([])
  
  // 当前激活的项目
  const activeProject = ref<Project | null>(null)
  
  // 加载状态
  const isLoading = ref(false)
  
  // 是否已初始化
  const isInitialized = ref(false)
  
  // ============== Getters ==============
  
  // 是否有激活的项目
  const hasActiveProject = computed(() => !!activeProject.value)
  
  // 项目数量
  const projectCount = computed(() => projectList.value.length)
  
  // 是否有多个项目
  const hasMultipleProjects = computed(() => projectList.value.length > 1)
  
  // 获取当前项目的人设配置
  const currentPersonaSettings = computed(() => {
    return activeProject.value?.persona_settings || DEFAULT_PERSONA_SETTINGS
  })
  
  // ============== Actions ==============
  
  /**
   * 初始化 - 从本地存储加载数据
   */
  function initialize() {
    if (isInitialized.value) return
    
    try {
      // 加载项目列表
      const storedList = uni.getStorageSync(PROJECT_LIST_KEY)
      if (storedList) {
        projectList.value = JSON.parse(storedList)
      }
      
      // 加载激活项目
      const storedActive = uni.getStorageSync(ACTIVE_PROJECT_KEY)
      if (storedActive) {
        activeProject.value = JSON.parse(storedActive)
      }
      
      isInitialized.value = true
    } catch (error) {
      console.error('Failed to load project data:', error)
    }
  }
  
  /**
   * 保存到本地存储
   */
  function saveToStorage() {
    try {
      uni.setStorageSync(PROJECT_LIST_KEY, JSON.stringify(projectList.value))
      if (activeProject.value) {
        uni.setStorageSync(ACTIVE_PROJECT_KEY, JSON.stringify(activeProject.value))
      } else {
        uni.removeStorageSync(ACTIVE_PROJECT_KEY)
      }
    } catch (error) {
      console.error('Failed to save project data:', error)
    }
  }
  
  /**
   * 获取项目列表
   */
  async function fetchProjects(): Promise<boolean> {
    isLoading.value = true
    
    try {
      const response = await get<{
        success: boolean
        projects: Project[]
        active_project_id: string | null
      }>('/api/projects')
      
      if (response.success && response.data) {
        projectList.value = response.data.projects || []
        
        // 如果有激活项目 ID，找到并设置
        if (response.data.active_project_id) {
          const active = projectList.value.find(
            p => p.id === response.data!.active_project_id
          )
          if (active) {
            activeProject.value = active
          }
        } else if (projectList.value.length === 1) {
          // 如果只有一个项目，自动激活
          activeProject.value = projectList.value[0]
        }
        
        saveToStorage()
        return true
      }
      
      return false
    } catch (error) {
      console.error('Failed to fetch projects:', error)
      return false
    } finally {
      isLoading.value = false
    }
  }
  
  /**
   * 创建新项目
   */
  async function createProject(data: ProjectCreateRequest): Promise<Project | null> {
    isLoading.value = true
    
    try {
      const response = await post<{
        success: boolean
        project: Project
      }>('/api/projects', data)
      
      if (response.success && response.data?.project) {
        const newProject = response.data.project
        projectList.value.unshift(newProject)
        
        // 如果是第一个项目，自动激活
        if (projectList.value.length === 1) {
          await setActiveProject(newProject)
        }
        
        saveToStorage()
        return newProject
      }
      
      return null
    } catch (error) {
      console.error('Failed to create project:', error)
      return null
    } finally {
      isLoading.value = false
    }
  }
  
  /**
   * 更新项目
   */
  async function updateProject(projectId: string, data: ProjectUpdateRequest): Promise<Project | null> {
    isLoading.value = true
    
    try {
      const response = await put<{
        success: boolean
        project: Project
      }>(`/api/projects/${projectId}`, data)
      
      if (response.success && response.data?.project) {
        const updatedProject = response.data.project
        
        // 更新列表中的项目
        const index = projectList.value.findIndex(p => p.id === projectId)
        if (index !== -1) {
          projectList.value[index] = updatedProject
        }
        
        // 如果是当前激活项目，同步更新
        if (activeProject.value?.id === projectId) {
          activeProject.value = updatedProject
        }
        
        saveToStorage()
        return updatedProject
      }
      
      return null
    } catch (error) {
      console.error('Failed to update project:', error)
      return null
    } finally {
      isLoading.value = false
    }
  }
  
  /**
   * 删除项目
   */
  async function deleteProject(projectId: string): Promise<boolean> {
    isLoading.value = true
    
    try {
      const response = await del<{ success: boolean }>(`/api/projects/${projectId}`)
      
      if (response.success) {
        // 从列表中移除
        projectList.value = projectList.value.filter(p => p.id !== projectId)
        
        // 如果删除的是当前激活项目
        if (activeProject.value?.id === projectId) {
          activeProject.value = projectList.value[0] || null
        }
        
        saveToStorage()
        return true
      }
      
      return false
    } catch (error) {
      console.error('Failed to delete project:', error)
      return false
    } finally {
      isLoading.value = false
    }
  }
  
  /**
   * 设置激活项目
   */
  async function setActiveProject(project: Project): Promise<boolean> {
    try {
      // 先更新本地状态
      activeProject.value = project
      saveToStorage()
      
      // 同步到后端
      const response = await post<{ success: boolean }>('/api/projects/switch', {
        project_id: project.id
      })
      
      return response.success
    } catch (error) {
      console.error('Failed to switch project:', error)
      // 即使后端失败，本地状态已更新
      return true
    }
  }
  
  /**
   * 清除所有项目数据
   */
  function clearProjects() {
    projectList.value = []
    activeProject.value = null
    try {
      uni.removeStorageSync(PROJECT_LIST_KEY)
      uni.removeStorageSync(ACTIVE_PROJECT_KEY)
    } catch (error) {
      console.error('Failed to clear project data:', error)
    }
  }
  
  /**
   * 生成带有人设上下文的系统提示词
   */
  function getPersonaSystemPrompt(): string {
    if (!activeProject.value) return ''
    
    const persona = activeProject.value.persona_settings
    const parts: string[] = []
    
    parts.push(`你现在扮演的是"${activeProject.value.name}"这个IP形象。`)
    
    if (persona.introduction) {
      parts.push(`IP简介：${persona.introduction}`)
    }
    
    if (persona.tone) {
      parts.push(`语气风格：${persona.tone}`)
    }
    
    if (persona.catchphrase) {
      parts.push(`口头禅：${persona.catchphrase}`)
    }
    
    if (persona.target_audience) {
      parts.push(`目标受众：${persona.target_audience}`)
    }
    
    if (persona.content_style) {
      parts.push(`内容风格：${persona.content_style}`)
    }
    
    if (persona.keywords && persona.keywords.length > 0) {
      parts.push(`常用关键词：${persona.keywords.join('、')}`)
    }
    
    if (persona.taboos && persona.taboos.length > 0) {
      parts.push(`内容禁忌（请避免提及）：${persona.taboos.join('、')}`)
    }
    
    return parts.join('\n')
  }
  
  // 初始化
  initialize()
  
  return {
    // State
    projectList,
    activeProject,
    isLoading,
    isInitialized,
    
    // Getters
    hasActiveProject,
    projectCount,
    hasMultipleProjects,
    currentPersonaSettings,
    
    // Actions
    initialize,
    fetchProjects,
    createProject,
    updateProject,
    deleteProject,
    setActiveProject,
    clearProjects,
    getPersonaSystemPrompt,
    saveToStorage
  }
})

// ============== 行业和语气选项 ==============

export const INDUSTRY_OPTIONS = [
  '通用',
  '医疗健康',
  '教育培训',
  '金融理财',
  '科技互联网',
  '电商零售',
  '餐饮美食',
  '旅游出行',
  '房产家居',
  '美妆护肤',
  '母婴育儿',
  '体育健身',
  '娱乐影视',
  '游戏动漫',
  '法律咨询',
  '职场成长',
  '情感心理',
  '三农乡村',
  '其他'
]

export const TONE_OPTIONS = [
  '专业亲和',
  '幽默风趣',
  '严肃正式',
  '温暖治愈',
  '犀利直接',
  '娓娓道来',
  '激情澎湃',
  '冷静理性'
]

File: backend/models/project.py
"""
Project Model - 项目（IP）数据模型

定义项目/IP的数据结构，用于多项目管理
"""

from datetime import datetime
from typing import Optional, List
from uuid import UUID, uuid4
from pydantic import BaseModel, Field


class PersonaSettings(BaseModel):
    """IP 人设配置"""
    tone: str = Field(default="专业亲和", description="语气风格：专业亲和/幽默风趣/严肃正式等")
    catchphrase: str = Field(default="", description="口头禅")
    target_audience: str = Field(default="", description="目标受众")
    benchmark_accounts: List[str] = Field(default_factory=list, description="对标账号列表")
    content_style: str = Field(default="", description="内容风格描述")
    taboos: List[str] = Field(default_factory=list, description="内容禁忌")
    keywords: List[str] = Field(default_factory=list, description="常用关键词")
    introduction: str = Field(default="", description="IP 简介")


class Project(BaseModel):
    """项目/IP 主模型"""
    id: UUID = Field(default_factory=uuid4, description="项目唯一标识")
    user_id: str = Field(..., description="关联用户ID")
    name: str = Field(..., description="项目名称，如'李医生科普IP'")
    industry: str = Field(default="通用", description="赛道，如'医疗健康'、'教育培训'等")
    avatar_letter: str = Field(default="", description="项目首字母/头像显示字符")
    avatar_color: str = Field(default="#3B82F6", description="头像背景色")
    persona_settings: PersonaSettings = Field(default_factory=PersonaSettings, description="人设配置")
    created_at: datetime = Field(default_factory=datetime.now, description="创建时间")
    updated_at: datetime = Field(default_factory=datetime.now, description="更新时间")
    is_active: bool = Field(default=True, description="是否为当前激活项目")
    
    class Config:
        json_encoders = {
            datetime: lambda v: v.isoformat(),
            UUID: lambda v: str(v)
        }


class ProjectCreate(BaseModel):
    """创建项目请求模型"""
    name: str = Field(..., min_length=1, max_length=50, description="项目名称")
    industry: str = Field(default="通用", description="赛道")
    persona_settings: Optional[PersonaSettings] = Field(default=None, description="人设配置")


class ProjectUpdate(BaseModel):
    """更新项目请求模型"""
    name: Optional[str] = Field(None, min_length=1, max_length=50, description="项目名称")
    industry: Optional[str] = Field(None, description="赛道")
    persona_settings: Optional[PersonaSettings] = Field(None, description="人设配置")


class ProjectSwitchRequest(BaseModel):
    """切换项目请求模型"""
    project_id: UUID = Field(..., description="要切换到的项目ID")


class ProjectListResponse(BaseModel):
    """项目列表响应模型"""
    success: bool = True
    projects: List[Project] = Field(default_factory=list, description="项目列表")
    active_project_id: Optional[UUID] = Field(None, description="当前激活的项目ID")


class ProjectResponse(BaseModel):
    """单个项目响应模型"""
    success: bool = True
    project: Project = Field(..., description="项目详情")


# 预设行业/赛道选项
INDUSTRY_OPTIONS = [
    "通用",
    "医疗健康",
    "教育培训",
    "金融理财",
    "科技互联网",
    "电商零售",
    "餐饮美食",
    "旅游出行",
    "房产家居",
    "美妆护肤",
    "母婴育儿",
    "体育健身",
    "娱乐影视",
    "游戏动漫",
    "法律咨询",
    "职场成长",
    "情感心理",
    "三农乡村",
    "其他"
]

# 预设语气风格选项
TONE_OPTIONS = [
    "专业亲和",
    "幽默风趣",
    "严肃正式",
    "温暖治愈",
    "犀利直接",
    "娓娓道来",
    "激情澎湃",
    "冷静理性"
]


